server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -207,7 +213,7 @@ protected DocumentParserContext(
             parent,
             dynamic,
             new HashSet<>(),
-            new HashSet<>(),
+            new HashSet<>(mappingLookup.fieldTypesLookup().getCopyToDestinationFields()),
```

Comment:
There is one existing user of this `DocumentParser#postProcessDynamicArrayMapping` but i think the intention of that code is the same as ours. It needs to have logic based on the fact that field is used as a copy_to destination. Previously that logic kicked in only if such a field was processed in this document already. With this, it will kick in the edge case of copy_to destination having actual values in the document. That seems to align with the intention of the code. 
cc @kderusso

Reply:
Yes, this is fine to only kick in if `copy_to` has values. Thank you for the ping.